### ISSUE TITLE
CommonJS/AMD/UMD compatibility

### ISSUE THREAD

**Issue by vvo** [algolia/algoliasearch-client-js#41]
## tl;dr;

```js
var algoliasearch = require('algoliasearch');
var client = algoliasearch({
  applicationID: '',
  apiKey: ''
});
```

## Boring stuff

Our current JavaScript client is not commonJS compatible.

It cannot be used with [requireJS](http://requirejs.org/) or [browserify](http://browserify.org/).

The only way to use our module is to copy paste the `<script src=..>` tag in a webpage and use the globally available `AlgoliaSearch`.

This type of JavaScript library usage pattern is old and we must provide modern module loaders compatibility.

## TODO

- [ ] module.exports = algoliasearch; (Why "algoliasearch"? see #40 )
- [ ] algoliasearch.helper = require('./lib/helper');
- [ ] for npm users: no build needed
- [ ] for requirejs/amd/umd/standalone users: build with [browserify standalone](https://github.com/substack/node-browserify#var-b--browserifyfiles-or-opts), `browserify -s algoliasearch index.js`
- [ ] add dist/ to .npmignore
- [ ] stop using grunt for build, browserify handles it all
- [ ] remove browserify-shim from package.json, no more used
